cmd/thanos/compact.go

```diff
@@ -41,6 +42,10 @@ func registerCompact(m map[string]setupFunc, app *kingpin.Application, name stri
 
 	retention := modelDuration(cmd.Flag("retention.default", "How long to retain samples in bucket. 0d - disables retention").Default("0d"))
 
+	retentionRes0 := modelDuration(cmd.Flag("retention.res0", "How long to retain raw samples in bucket. 0d - disables this retention").Default("0d"))
```

Comment:
can we make `res` a full world, with actuall time? Would be more verbose.

so `retention.resolution-raw` `retention.resolution-5m` etc

Reply:
Should be totally fine and better for users :-)